### PR TITLE
Test add remove replica

### LINKED ISSUE #57: Replication: Add/Remove replicas testing
Add automated tests for adding and removing replicas in a replication deployment

### PR DESCRIPTION
**Pull request by MuhammadQadora**
fix #57 

Tested the addition and removal of a new replica and the persistence of the data before and after the newly added replica became master.

## Summary by CodeRabbit

- **New Features**
	- Introduced new jobs in the testing framework for managing the addition and removal of replicas in both single-zone and multi-zone configurations.
	- Added new configuration option "3" for service parameters, enhancing flexibility for users.
	- Launched a comprehensive testing script for validating replication instances and failover functionalities.

- **Chores**
	- Added `curlify` dependency to facilitate conversion of cURL commands into Python requests.

- **Tests**
	- Introduced timeout parameters to various test functions for improved deployment control.
	- Enhanced API interaction resilience by increasing timeout settings across multiple methods.

### CODE REVIEW

**Comment by MuhammadQadora** [comment]
Hi,
the most significant finding for this test was the need to change the payload (data) when requesting to update the parameters (replica count) from {"numReplica": "3"} to {"requestParams": {"numReplica": 3 }}. Here we had an ambiguous response from the API regarding a value error, which turned out after speaking with Omnistrate to be an integer value issue, 3 instead of "3". @dudizimber asked Omnistrate to try and change such responses to somthing meaningful.
As with other tests ,I received an API request timeout, and had to increase the request timeout from 15 seconds to 60 seconds. For the time being I have returned it to 15 seconds.

File: falkordb-node/node-entrypoint.sh [hunk 1]
```diff
@@ -409,45 +409,46 @@ if [ "$RUN_NODE" -eq "1" ]; then
...
-  else
-    echo "port $SENTINEL_PORT" >>$SENTINEL_CONF_FILE
-  fi
-
-  redis-server $SENTINEL_CONF_FILE --sentinel --logfile $SENTINEL_LOG_FILE_PATH &
-  sentinel_pid=$!
-  tail -F $SENTINEL_LOG_FILE_PATH &
-
-  sleep 10
+  if [[ "$NODE_INDEX" == "0" || "$NODE_INDEX" == "1" ]];then
```
Comment (dudizimber) [inline falkordb-node/node-entrypoint.sh:412]: If all the code inside the first condition is also inside the second condition, why don't you merge them and clean the code?

File: omnistrate_tests/test_replication_replicas.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            falkordb_user="falkordb",
+            falkordb_password="falkordb",
+            nodeInstanceType=args.instance_type,
+            storageSize=args.storage_size,
+            enableTLS=args.tls,
+            RDBPersistenceConfig=args.rdb_config,
+            AOFPersistenceConfig=args.aof_config,
+        )
+
+        logging.info('The code will now run the add_data function')
```
Comment (dudizimber) [inline omnistrate_tests/test_replication_replicas.py:114]: Are these logs necessary? with the Info severity?

File: omnistrate_tests/test_replication_replicas.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+
+parser.add_argument("--instance-name", required=True)
+parser.add_argument("--instance-description", required=False, default="test-standalone")
+parser.add_argument("--instance-type", required=True)
+parser.add_argument("--storage-size", required=False, default="30")
+parser.add_argument("--tls", action="store_true")
+parser.add_argument("--rdb-config", required=False, default="medium")
+parser.add_argument("--aof-config", required=False, default="always")
+parser.add_argument("--replica-count", required=False, default="1")
```
Comment (dudizimber) [inline omnistrate_tests/test_replication_replicas.py:45]: The api param replicacount has only 2 or 3 as options. Isn't it better to match the meaning, and force here to be 2 at minimum?

File: omnistrate_tests/test_replication_replicas.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+    endpoint = instance.get_cluster_endpoint()
+
+    id_key = "sz" if args.resource_key == "single-Zone" else "mz"
+
+    try:
+        client = Redis(
+        host=f"{endpoint["endpoint"]}", port=endpoint['ports'][0],
+        username="falkordb", # use your Redis user. More info https://redis.io/docs/latest/operate/oss_and_stack/management/security/acl/
+        password="falkordb", # use your Redis password
```
Comment (dudizimber) [inline omnistrate_tests/test_replication_replicas.py:156]: Is the comment necessary?

**Review by dudizimber — CHANGES_REQUESTED** [review]
see comments

**Review by dudizimber — DISMISSED** [review]

**Review by dudizimber — CHANGES_REQUESTED** [review]
1. Instance types during testing must be `e2-medium`.
2. Must use random passwords when creating the falkordb instance

File: omnistrate_tests/test_replication_replicas.py [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+    )
+    service_model = omnistrate.get_service_model(
+        args.service_id, product_tier.service_model_id
+    )
+
+    logging.info(f"Product tier id: {product_tier.product_tier_id} for {args.ref_name}")
+
+    instance = omnistrate.instance(
+        service_id=args.service_id,
+        service_provider_id="sp-JvkxkPhinN",
```
Comment (dudizimber) [inline omnistrate_tests/test_replication_replicas.py:93]: Why not from args?

File: omnistrate_tests/test_replication_replicas.py [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+    try:
+        password = randbytes(16).hex()
+        instance.create(
+            wait_for_ready=True,
+            deployment_cloud_provider=args.cloud_provider,
+            deployment_region=args.region,
+            name=args.instance_name,
+            description=args.instance_description,
+            falkordb_user="falkordb",
+            falkordb_password="falkordb",
```
Comment (dudizimber) [inline omnistrate_tests/test_replication_replicas.py:116]: Why are we not using the random password

**Review by dudizimber — CHANGES_REQUESTED** [review]
see comments

**Review by dudizimber — APPROVED** [review]

**Pull request merged**